lib/heborn_migration/web/controllers/page_controller.ex

```diff
@@ -1,7 +1,25 @@
 defmodule HEBornMigration.Web.PageController do
   use HEBornMigration.Web, :controller
 
+  alias HEBornMigration.Controller.Account, as: AccountController
+  alias HEBornMigration.Model.Claim
+
   def index(conn, _params) do
     render conn, "index.html"
   end
+
+  def claim(conn, params) do
+    display_name = Map.fetch!(params, "display_name")
+
+    case AccountController.claim(display_name) do
+      {:ok, token} ->
+        json conn, %{token: token}
+      {:error, changeset} ->
+        data = Claim.format_error(changeset)
+
+        conn
+        |> put_status(422)
+        |> json(%{errors: data})
+    end
+  end
```

Comment:
pretty simple, huh?

422 seems the most suitable error code for invalid input:

> The 422 (Unprocessable Entity) status code means the server
> understands the content type of the request entity - hence a 415
> (Unsupported Media Type) status code is inappropriate-, and the
> syntax of the request entity is correct - thus a 400 (Bad Request)
> status code is inappropriate - but was unable to process the
> contained instructions.
> 
> For example, this error condition may occur if an XML request
> body contains well-formed (i.e., syntactically correct), but
> semantically erroneous, XML instructions.